ShowerShow/Control/AchievementController.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+using ShowerShow.DTO;
+using ShowerShow.Models;
+using ShowerShow.Repository.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ShowerShow.Control
+{
+    internal class AchievementController
+    {
+        private readonly ILogger<AchievementController> _logger;
+        private readonly IAchievementService _achievementService;
+
+        public AchievementController(ILogger<AchievementController> logger, IAchievementService service)
+        {
+            _logger = logger;
+            _achievementService = service;
+        }
+
+        [Function(nameof(GetAchievementsById))] 
+        [OpenApiOperation(operationId: "GetUserAchievements", tags: new[] { "Achievement" })]
+        [OpenApiParameter(name: "UserId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The User ID parameter")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(List<Achievement>), Description = "The OK response with all achievements from user.")]
+        public async Task<IActionResult> GetAchievementsById([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "user/{UserId}/achievements")] HttpRequestData req, Guid UserId)
```

Comment:
IactionResult has to change to HttpResponseData as a return type for all endpoints now. Its a recent development